hoot-core/src/main/cpp/hoot/core/algorithms/linearreference/WaySublineRemover.cpp

```diff
@@ -60,10 +63,28 @@ std::vector<ElementId> WaySublineRemover::removeSubline(
   if (subline.getStart().isExtreme() && subline.getEnd().isExtreme())
   {
     LOG_TRACE(
-      "Subline matches covers entire way; removing entire way: " << way->getElementId());
-    // Use RemoveElementByEid here instead of RecursiveElementRemover so that the way is removed
-    // from its parent before its removal.
-    RemoveElementByEid(way->getElementId()).apply(map);
+      "Subline matches covers entire way. Removing entire way: " << way->getElementId() << "...");
+
+    if (removeWholeWayRecursively)
+    {
+      // remove it from parents and the map
+      RemoveElementByEid(way->getElementId()).apply(map);
+      // remove its children
+      RecursiveElementRemover(way->getElementId()).apply(map);
+    }
+    else
+    {
+      // Use RemoveElementByEid here instead of RecursiveElementRemover so that the way is removed
+      // from its parent before its removal. We also don't want to remove RecursiveElementRemover here
+      // b/c the way's nodes may still belong to one the temp relations we created during splitting.
+      RemoveElementByEid(way->getElementId()).apply(map);
+    }
+
```

Comment:
This part got a little confusing. I think most of the time when making a single call for subline removal, you'd want to remove all the child way nodes as well. During merging, though, I ran into situations where shared way nodes were being deleted to early. So for diff conflate merging, I'm removing those shared nodes at the very end.